### PR TITLE
cosmosdb - add throughput to remaining resources

### PR DESCRIPTION
**Pull request by katbyte**
continuation of #4616

### CODE REVIEW

File: azurerm/resource_arm_cosmosdb_sql_container.go [hunk 1]
```diff
@@ -158,6 +174,75 @@ func resourceArmCosmosDbSQLContainerCreate(d *schema.ResourceData, meta interfac
...
+				Resource: &documentdb.ThroughputResource{
+					Throughput: utils.Int32(int32(d.Get("throughput").(int))),
+				},
+			},
+		}
+
+		throughputFuture, err := client.UpdateSQLContainerThroughput(ctx, id.ResourceGroup, id.Account, id.Database, id.Container, throughputParameters)
+		if err != nil {
+			if throughputFuture.Response().StatusCode == http.StatusNotFound {
+				d.Set("throughput", nil)
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmosdb_sql_container.go]: do we need this?

File: azurerm/resource_arm_cosmosdb_sql_container.go [hunk 2]
```diff
@@ -158,6 +174,75 @@ func resourceArmCosmosDbSQLContainerCreate(d *schema.ResourceData, meta interfac
...
+			ThroughputUpdateProperties: &documentdb.ThroughputUpdateProperties{
+				Resource: &documentdb.ThroughputResource{
+					Throughput: utils.Int32(int32(d.Get("throughput").(int))),
+				},
+			},
+		}
+
+		throughputFuture, err := client.UpdateSQLContainerThroughput(ctx, id.ResourceGroup, id.Account, id.Database, id.Container, throughputParameters)
+		if err != nil {
+			if throughputFuture.Response().StatusCode == http.StatusNotFound {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmosdb_sql_container.go]: should this be `response.WasNotFound`?

File: azurerm/resource_arm_cosmosdb_cassandra_keyspace.go [hunk 3]
```diff
@@ -110,6 +119,60 @@ func resourceArmCosmosDbCassandraKeyspaceCreate(d *schema.ResourceData, meta int
...
+				Resource: &documentdb.ThroughputResource{
+					Throughput: utils.Int32(int32(d.Get("throughput").(int))),
+				},
+			},
+		}
+
+		throughputFuture, err := client.UpdateCassandraKeyspaceThroughput(ctx, id.ResourceGroup, id.Account, id.Keyspace, throughputParameters)
+		if err != nil {
+			if throughputFuture.Response().StatusCode == http.StatusNotFound {
+				d.Set("throughput", nil)
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmosdb_cassandra_keyspace.go]: do we need this?

File: azurerm/resource_arm_cosmosdb_cassandra_keyspace.go [hunk 4]
```diff
@@ -110,6 +119,60 @@ func resourceArmCosmosDbCassandraKeyspaceCreate(d *schema.ResourceData, meta int
...
+			ThroughputUpdateProperties: &documentdb.ThroughputUpdateProperties{
+				Resource: &documentdb.ThroughputResource{
+					Throughput: utils.Int32(int32(d.Get("throughput").(int))),
+				},
+			},
+		}
+
+		throughputFuture, err := client.UpdateCassandraKeyspaceThroughput(ctx, id.ResourceGroup, id.Account, id.Keyspace, throughputParameters)
+		if err != nil {
+			if throughputFuture.Response().StatusCode == http.StatusNotFound {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmosdb_cassandra_keyspace.go]: should this be utils/response.wasnotfound?

File: azurerm/resource_arm_cosmosdb_sql_database.go [hunk 5]
```diff
@@ -110,6 +126,60 @@ func resourceArmCosmosDbSQLDatabaseCreate(d *schema.ResourceData, meta interface
...
+			ThroughputUpdateProperties: &documentdb.ThroughputUpdateProperties{
+				Resource: &documentdb.ThroughputResource{
+					Throughput: utils.Int32(int32(d.Get("throughput").(int))),
+				},
+			},
+		}
+
+		throughputFuture, err := client.UpdateSQLDatabaseThroughput(ctx, id.ResourceGroup, id.Account, id.Database, throughputParameters)
+		if err != nil {
+			if throughputFuture.Response().StatusCode == http.StatusNotFound {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmosdb_sql_database.go]: could this be utils/response.wasnotfound?

File: azurerm/resource_arm_cosmosdb_sql_database.go [hunk 6]
```diff
@@ -110,6 +126,60 @@ func resourceArmCosmosDbSQLDatabaseCreate(d *schema.ResourceData, meta interface
...
+				d.Set("throughput", nil)
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmosdb_sql_database.go]: do we need this?

File: azurerm/resource_arm_cosmosdb_table.go [hunk 7]
```diff
@@ -110,6 +126,60 @@ func resourceArmCosmosDbTableCreate(d *schema.ResourceData, meta interface{}) er
...
+			ThroughputUpdateProperties: &documentdb.ThroughputUpdateProperties{
+				Resource: &documentdb.ThroughputResource{
+					Throughput: utils.Int32(int32(d.Get("throughput").(int))),
+				},
+			},
+		}
+
+		throughputFuture, err := client.UpdateTableThroughput(ctx, id.ResourceGroup, id.Account, id.Table, throughputParameters)
+		if err != nil {
+			if throughputFuture.Response().StatusCode == http.StatusNotFound {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmosdb_table.go]: could this be utils/response.wasnotfound?

File: azurerm/resource_arm_cosmosdb_table.go [hunk 8]
```diff
@@ -110,6 +126,60 @@ func resourceArmCosmosDbTableCreate(d *schema.ResourceData, meta interface{}) er
...
+				d.Set("throughput", nil)
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmosdb_table.go]: do we need this?

**Comment by brandonmbanks** [comment]
Thank you for this! 👍

**Pull request merged**